src/components/menu/Menu.tsx

```diff
@@ -187,36 +186,31 @@ const Menu: GenericComponent<MenuProps> = ({
 
   const show = () => {
     onShow && onShow();
-
-    setActive(true);
     addEvents();
   };
 
   const hide = () => {
     onHide && onHide();
-
-    setActive(false);
     removeEvents();
   };
 
   useEffect(() => {
     const { width, height } = menuNode?.current?.getBoundingClientRect() || {};
 
+    active ? show() : hide();
     width && setWidth(width);
     height && setHeight(height);
-  }, [menuNode]);
 
-  useEffect(() => {
-    stateActive ? show() : hide();
+    return () => {
+      active && removeEvents();
+    };
+  }, [active, menuNode]);
```

Comment:
Should these be 2 different useEffects? (1 for when active changes and 1 for when menuNode changes)
Otherwise it seems like it's possible the events get added twice if the menuNode happens to change

Reply:
I think the menuNode doesn't change, except on initial render 🤔. But just in case it happens, I can try and move it to a seperate useEffect :). Thanks for the remark 🙌